### PR TITLE
Shape copy/conversion helper

### PR DESCRIPTION
**Pull request by ash2k**

### CODE REVIEW

File: pkg/orchestration/wiring/ec2compute/common/common.go [hunk 1]
```diff
@@ -140,7 +140,12 @@ func WireUp(microServiceNameInSpec, ec2ComputePlanName string, stateResource *or
 	for _, dependency := range dependencies {
 		bound := false
 		if bindableShape, ok := dependency.Contract.FindShape(knownshapes.BindableEnvironmentVariablesShape); ok {
-			resourceReference := bindableShape.(*knownshapes.BindableEnvironmentVariables).Data.ServiceInstanceName
+			typedBindableShape := &knownshapes.BindableEnvironmentVariables{}
```
Comment (kopper) [inline pkg/orchestration/wiring/ec2compute/common/common.go]: This code belongs to knownshapes package, so it is easier for clients to use it. Instead of 

`dependency.Contract.FindShape...` 

I'd prefer

`bindableShapeData := knownshapes.FindBindableEnvironmentVariablesShape(dependency.Contract)`

`FindBindableEnvironmentVariablesShape` should return `BindableEnvironmentVariablesDate`

File: pkg/orchestration/wiring/wiringplugin/shapes.go [hunk 2]
```diff
@@ -39,3 +46,30 @@ type ShapeMeta struct {
...
+		// Same type already, just deep copy
+		sv := reflect.ValueOf(source)
+		tv := reflect.ValueOf(target)
+		method := sv.MethodByName("DeepCopyInto")
+		method.Call([]reflect.Value{tv})
+	default:
+		// Convert by round-tripping to and from JSON
+		data, err := json.Marshal(source)
+		if err != nil {
+			return errors.WithStack(err)
```
Comment (kopper) [inline pkg/orchestration/wiring/wiringplugin/shapes.go]: wrap these errors with message to make it more obvious if marshalling or unmarshalling failed

**Comment by ash2k** [comment]
ready for re-review

File: .bazelrc [hunk 3]
```diff
@@ -13,7 +13,6 @@ build:ci --experimental_sandbox_base=/dev/shm
 # Pass -test.v to each test binary to print the test names
 test:ci --test_arg="-test.v"
 test:ci --test_output=all
-test:ci --test_verbose_timeout_warnings
```
Comment (brjacobsatlassian) [inline .bazelrc]: This doesn't look related

File: pkg/orchestration/wiring/wiringplugin/shapes.go [hunk 4]
```diff
@@ -39,3 +46,30 @@ type ShapeMeta struct {
...
+	switch {
+	case st == tt:
+		// Same type already, just deep copy
+		sv := reflect.ValueOf(source)
+		tv := reflect.ValueOf(target)
+		method := sv.MethodByName("DeepCopyInto")
+		method.Call([]reflect.Value{tv})
+	default:
+		// Convert by round-tripping to and from JSON
+		data, err := json.Marshal(source)
```
Comment (brjacobsatlassian) [inline pkg/orchestration/wiring/wiringplugin/shapes.go]: Perhaps we should look for a specific converter first?

File: .bazelrc (hunk 3 above)
Comment (ash2k) [reply .bazelrc]: It's in a separate commit

File: pkg/orchestration/wiring/wiringplugin/shapes.go (hunk 4 above)
Comment (ash2k) [reply pkg/orchestration/wiring/wiringplugin/shapes.go]: What do you mean?

Comment (brjacobsatlassian) [reply pkg/orchestration/wiring/wiringplugin/shapes.go]: It'd be nice if there were a way to define a map(from, to) function instead of falling back to serializing and then deserializing to convert any different shapes.

Comment (ash2k) [reply pkg/orchestration/wiring/wiringplugin/shapes.go]: Yes, that is how scheme `k8s.io/apimachinery/pkg/runtime.Scheme` works. I thought about it but I think it would be overkill. At least for now we don't need such mechanism, and we can build it later. I've also thought about using `k8s.io/apimachinery/pkg/runtime.DefaultUnstructuredConverter` somehow but there are some potential issues with using it (name of the field in struct not matching the name of the field in JSON) so I stopped looking into it.
